app/views/repositories/_navigation.html.erb

```diff
@@ -30,37 +30,44 @@ See doc/COPYRIGHT.rdoc for more details.
 <% content_for :header_tags do %>
   <%= javascript_include_tag 'repository_navigation' %>
 <% end %>
-
-<% content_for :toolbar do %>
-  <li class="toolbar-item">
-    <%= link_to stats_project_repository_path(@project), class: 'button' do %>
-      <i class="icon icon-stats1"></i> <%= l(:label_statistics) %>
-    <% end %>
-  </li>
-  <%# rev => nil prevents overwriting the rev parameter queried for in the form with the parameter from the request %>
-  <%= form_tag({:action => controller.action_name, :project_id => @project, :path => to_path_param(@path), :rev => nil}, {:method => :get, :id => 'revision_selector'}) do -%>
-    <!-- Branches Dropdown -->
-    <% if !@repository.branches.nil? && @repository.branches.length > 0 -%>
+<div class="toolbar-container">
+  <div id="toolbar">
+    <div class="title-container">
```

Comment:
What is the reason for not using the `toolbar` helper that you introduced, here?

Something along those lines:

````
<%= toolbar title: render :partial => 'breadcrumbs', :locals => { :path => @path, :revision => @rev }.merge(merge) do %>
````